src/main/java/com/google/sps/servlets/BookmarkHandlerServlet.java

```diff
@@ -0,0 +1,139 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.*;
+import com.google.sps.data.Bookmark;
+import com.google.sps.storage.BookmarkStorageInterface;
+import com.google.sps.storage.BookmarkStorageManager;
+import com.google.sps.storage.BookmarkStorageException;
+
+
+/** Servlet that interacts with setting and fetching bookmarks */
+@WebServlet("/bookmark")
+public class BookmarkHandlerServlet extends HttpServlet {
+
+    private static final String REQUEST_EMAIL_PARAM = "email";
+    private static final String REQUEST_VIDEO_ID_PARAM = "videoId";
+    private static final String REQUEST_TIMESTAMP_PARAM = "timestamp";
+    private static final String REQUEST_TITLE_PARAM = "title";
+    private static final String REQUEST_CONTENT_PARAM = "content";
+    private static final String REQUEST_BOOKMARK_ID_PARAM = "bookmarkId";
+    private static final String REQUEST_FUNCTION_PARAM = "function";
+    private static final String REQUEST_ADD_FUNCTION = "add";
+    private static final String REQUEST_REMOVE_FUNCTION = "remove";
+    private static final String RESPONSE_JSON_CONTENT = "json";
+
+    private BookmarkStorageInterface storage;
+
+    /**
+     * Gets the bookmarks of the current video and user
+     * @param request The request object 
+     * @param response The response object
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String email = (String) request.getParameter(REQUEST_EMAIL_PARAM);
+        String videoId = (String) request.getParameter(REQUEST_VIDEO_ID_PARAM);
+        response.setContentType(RESPONSE_JSON_CONTENT);
+
+        // Tries to get all the bookmarks for the current user and video
+        if (storage == null) {
+            storage = new BookmarkStorageManager();
+        }
+        List<Bookmark> bookmarks = new ArrayList<>();
+        try {
+            bookmarks = storage.getAllBookmarks(email, videoId);
+        } catch (BookmarkStorageException e) {
+            response.getWriter().println(e.getReasonAsJsonString());
+            return;
+        }
+
+        // Converts the fetched database data and passes it to the front end
+        Gson gson = new Gson();
```

Comment:
Nit: Let's move gson to a member var so we do need to init in different functions

Reply:
addressed